Error occurs when building exported mbed-os programs on e2studio

### Description

I faced the several build error when compiled the exported **mbed-os programs** on Renesas' IDE **e2studio**.
The build errors are a bit different by programs, but I frequently look messages such as "No such file or directory".

The procedure is as follows:
1. Export the program on Online Compiler
2. Build the exported program on e2studio

The error that is shown when building exported `mbed-os-example-blinky(mbed os 5.8.5)` on e2studio is as follows:
```
../mbed-os/features/storage/FEATURE_STORAGE/storage-volume-manager/source/storage_volume.cpp:25:59: fatal error: storage-volume-manager/storage_volume_manager.h: No such file or directory
 #include "storage-volume-manager/storage_volume_manager.h"
                                                           ^
../mbed-os/features/storage/FEATURE_STORAGE/storage-volume-manager/source/storage_volume_manager.cpp:25:59: fatal error: storage-volume-manager/storage_volume_manager.h: No such file or directory
 #include "storage-volume-manager/storage_volume_manager.h"
                                                           ^
../mbed-os/features/storage/FEATURE_STORAGE/flash-journal/flash-journal-strategy-sequential/flash_journal_crc.c:18:65: fatal error: flash-journal-strategy-sequential/flash_journal_crc.h: No such file or directory
 #include "flash-journal-strategy-sequential/flash_journal_crc.h"
                                                                 ^
make: *** [mbed-os/features/storage/FEATURE_STORAGE/storage-volume-manager/source/storage_volume_manager.o] Error 1
mbed-os/features/storage/FEATURE_STORAGE/storage-volume-manager/source/subdir.mk:21: recipe for target 'mbed-os/features/storage/FEATURE_STORAGE/storage-volume-manager/source/storage_volume_manager.o' failed
make: *** Waiting for unfinished jobs....
make: *** [mbed-os/features/storage/FEATURE_STORAGE/storage-volume-manager/source/storage_volume.o] Error 1
mbed-os/features/storage/FEATURE_STORAGE/storage-volume-manager/source/subdir.mk:21: recipe for target 'mbed-os/features/storage/FEATURE_STORAGE/storage-volume-manager/source/storage_volume.o' failed
make: *** [mbed-os/features/storage/FEATURE_STORAGE/flash-journal/flash-journal-strategy-sequential/flash_journal_crc.o] Error 1
mbed-os/features/storage/FEATURE_STORAGE/flash-journal/flash-journal-strategy-sequential/subdir.mk:24: recipe for target 'mbed-os/features/storage/FEATURE_STORAGE/flash-journal/flash-journal-strategy-sequential/flash_journal_crc.o' failed
../mbed-os/features/storage/FEATURE_STORAGE/flash-journal/flash-journal-strategy-sequential/strategy.c:18:65: fatal error: flash-journal-strategy-sequential/flash_journal_crc.h: No such file or directory
 #include "flash-journal-strategy-sequential/flash_journal_crc.h"
                                                                 ^
make: *** [mbed-os/features/storage/FEATURE_STORAGE/flash-journal/flash-journal-strategy-sequential/strategy.o] Error 1
mbed-os/features/storage/FEATURE_STORAGE/flash-journal/flash-journal-strategy-sequential/subdir.mk:24: recipe for target 'mbed-os/features/storage/FEATURE_STORAGE/flash-journal/flash-journal-strategy-sequential/strategy.o' failed

```

With respect to file of "No such file or directory", it seems that file was exported but not included headers in the include paths.

Anyway, I think that this should **not** be the expected behavior, could you confirm the above? 



### Issue request type
<!-- 
    Required
    Please add only one X to one of the following types. Do not fill multiple types. (Split the issue otherwise.)
    Please note this is not a GitHub task list; indenting the boxes or changing the format to add a '.' or '*' in front
    of them changes the meaning incorrectly. The only changes to make are to add a description under the
    description heading and to add an 'x' to the correct box.

    [] Question
    [ ] Enhancement 
    [ ] Bug 
-->

[ ] Question  
[ ] Enhancement  
[X] Bug

@toyowata @theotherjimmy 

I verified Export by following two patterns.
<br><br>
**By Online Compiler**
As shown the above, I faced the "No such file or directory" error when building the exported program.
Since the corresponding files(Following 5 passes) look like **unnecessary include**, I set them outside of build target on e2studio as a trial.
```
features/cryptocell
features/FEATURE_BLE
features/FEATURE_UVISOR
features/nanostack
features/storage
```
As the result of it, the following build error was shown.
```
arm-none-eabi-g++ @"mbed-os-example-blinky.elf.in"
arm-none-eabi-g++: error: unrecognized command line option '--wrap=main'; did you mean '--warn-main'?
arm-none-eabi-g++: error: unrecognized command line option '--wrap=_malloc_r'
arm-none-eabi-g++: error: unrecognized command line option '--wrap=_free_r'
arm-none-eabi-g++: error: unrecognized command line option '--wrap=_realloc_r'; did you mean '--warn-realloc-lhs'?
arm-none-eabi-g++: error: unrecognized command line option '--wrap=_memalign_r'
arm-none-eabi-g++: error: unrecognized command line option '--wrap=_calloc_r'
arm-none-eabi-g++: error: unrecognized command line option '--wrap=exit'; did you mean '--graphite'?
arm-none-eabi-g++: error: unrecognized command line option '--wrap=atexit'; did you mean '--graphite'?
makefile:130: recipe for target 'mbed-os-example-blinky.elf' failed
make: *** [mbed-os-example-blinky.elf] Error 1
```


**By Mbed CLI**
I built the exported program by running the following command.
```
$ mbed import http://os.mbed.com/teams/mbed-os-examples/code/mbed-os-example-blinky/
$ cd mbed-os-example-blinky
$ mbed export -m RZ_A1H -i e2studio
Build the exported program on e2studio
```
As the result of it, the following build error was shown("No such file or directory" error is not shown and the same "--wrap" error as above is shown).
```
arm-none-eabi-g++ @"mbed-os-example-blinky.elf.in"
arm-none-eabi-g++: error: unrecognized command line option '--wrap=main'; did you mean '--warn-main'?
arm-none-eabi-g++: error: unrecognized command line option '--wrap=__malloc_r'
arm-none-eabi-g++: error: unrecognized command line option '--wrap=__free_r'
arm-none-eabi-g++: error: unrecognized command line option '--wrap=__realloc_r'
arm-none-eabi-g++: error: unrecognized command line option '--wrap=__memalign_r'
arm-none-eabi-g++: error: unrecognized command line option '--wrap=__calloc_r'
arm-none-eabi-g++: error: unrecognized command line option '--wrap=exit'; did you mean '--graphite'?
arm-none-eabi-g++: error: unrecognized command line option '--wrap=atexit'; did you mean '--graphite'?
makefile:185: recipe for target 'mbed-os-example-blinky.elf' failed
make: *** [mbed-os-example-blinky.elf] Error 1
```


To summarize the above, I think that check point is the below.<br>
- When exporting from Online Compiler, the **unnecessary include path** seems to set.<br>
- Whichever export method is used, the **--wrap** option related problem seems to be blocking the build on e2studio.<br>

--- a/tools/export/e2studio/.cproject.tmpl
+++ b/tools/export/e2studio/.cproject.tmpl
@@ -326,15 +326,18 @@
 									{% endfor %}
 								</option>
 								<option id="ilg.gnuarmeclipse.managedbuild.cross.option.cpp.linker.libs.{{u.id}}" superClass="ilg.gnuarmeclipse.managedbuild.cross.option.cpp.linker.libs" valueType="libs">
-									{% for lib in opts['ld']['user_libraries'] %}
+									{% for lib in opts['ld']['system_libraries'] %}
 									<listOptionValue builtIn="false" value="{{lib}}"/>
 									{% endfor %}
-									{% for lib in opts['ld']['system_libraries'] %}
+									{% for lib in opts['ld']['user_libraries'] %}
 									<listOptionValue builtIn="false" value="{{lib}}"/>
 									{% endfor %}
 								</option>
 								<option id="ilg.gnuarmeclipse.managedbuild.cross.option.cpp.linker.otherobjs.{{u.id}}" superClass="ilg.gnuarmeclipse.managedbuild.cross.option.cpp.linker.otherobjs" valueType="userObjs">
 									{% for path in opts['ld']['object_files'] %}
+									<listOptionValue builtIn="false" value="&quot;${ProjDirPath}/{{path}}&quot;"/>
+									{% endfor %}
+									{% for path in opts['ld']['user_library_files'] %}
 									<listOptionValue builtIn="false" value="&quot;${ProjDirPath}/{{path}}&quot;"/>
 									{% endfor %}
 								</option>

--- a/tools/export/gnuarmeclipse/.cproject.tmpl
+++ b/tools/export/gnuarmeclipse/.cproject.tmpl
@@ -326,15 +326,18 @@
 									{% endfor %}
 								</option>
 								<option id="ilg.gnuarmeclipse.managedbuild.cross.option.cpp.linker.libs.{{u.id}}" superClass="ilg.gnuarmeclipse.managedbuild.cross.option.cpp.linker.libs" valueType="libs">
-									{% for lib in opts['ld']['user_libraries'] %}
+									{% for lib in opts['ld']['system_libraries'] %}
 									<listOptionValue builtIn="false" value="{{lib}}"/>
 									{% endfor %}
-									{% for lib in opts['ld']['system_libraries'] %}
+									{% for lib in opts['ld']['user_libraries'] %}
 									<listOptionValue builtIn="false" value="{{lib}}"/>
 									{% endfor %}
 								</option>
 								<option id="ilg.gnuarmeclipse.managedbuild.cross.option.cpp.linker.otherobjs.{{u.id}}" superClass="ilg.gnuarmeclipse.managedbuild.cross.option.cpp.linker.otherobjs" valueType="userObjs">
 									{% for path in opts['ld']['object_files'] %}
+									<listOptionValue builtIn="false" value="&quot;${ProjDirPath}/{{path}}&quot;"/>
+									{% endfor %}
+									{% for path in opts['ld']['user_library_files'] %}
 									<listOptionValue builtIn="false" value="&quot;${ProjDirPath}/{{path}}&quot;"/>
 									{% endfor %}
 								</option>

--- a/tools/export/gnuarmeclipse/__init__.py
+++ b/tools/export/gnuarmeclipse/__init__.py
@@ -90,7 +90,9 @@ class GNUARMEclipse(Exporter):
         # TODO: use some logger to display additional info if verbose
 
         libraries = []
+        library_files = []
         for lib in self.libraries:
+            library_files.append(self.filter_dot(lib))
             l, _ = splitext(basename(lib))
             libraries.append(l[3:])
 
@@ -178,6 +180,7 @@ class GNUARMEclipse(Exporter):
 
             opts['ld']['object_files'] = objects
             opts['ld']['user_libraries'] = libraries
+            opts['ld']['user_library_files'] = library_files
             opts['ld']['system_libraries'] = self.system_libraries
             opts['ld']['script'] = join(id.capitalize(),
                                         "linker-script-%s.ld" % id)

